cmd/snap-bootstrap/cmd_initramfs_mounts.go

```diff
@@ -551,13 +553,13 @@ func unlockIfEncrypted(name string, last bool) (string, error) {
 			// TODO:UC20: we might want some better error handling here - eg, if tpmErr is a
 			//            *os.PathError returned from go-tpm2 then this is an indicator that there
 			//            is no TPM device. But other errors probably shouldn't be ignored.
-			if last && tpmErr == nil {
+			if lockKeyOnFinish && tpmErr == nil {
 				// Lock access to the sealed keys. This should be called whenever there
 				// is a TPM device detected, regardless of whether secure boot is enabled
 				// or there is an encrypted volume to unlock. Note that snap-bootstrap can
 				// be called several times during initialization, and if there are multiple
 				// volumes to unlock we should lock access to the sealed keys only after
-				// the last encrypted volume is unlocked.
+				// the lockKeyOnFinish encrypted volume is unlocked.
```

Comment:
... after the last encrypted volume is unlocked,  in which case lockKeyOnFinish should be set to true.